src/main/java/org/opensearch/performanceanalyzer/rca/store/rca/hot_node/ThreadMetricsRca.java

```diff
@@ -0,0 +1,385 @@
+                if (newThreadNames.contains(entry.getKey())) {
+                    continue;
+                }
+                pruneExpiredEntries(timestamp, entry.getValue());
+            }
+            metricsByThreadName.entrySet().removeIf(e -> e.getValue().size() == 0);
+        }
+
+        private void pruneExpiredEntries(long endTimeStamp, Deque<ThreadMetric> windowDeque) {
+            while (!windowDeque.isEmpty()
+                    && TimeUnit.MILLISECONDS.toSeconds(
+                                    endTimeStamp - windowDeque.peekLast().getTimeStamp())
+                            > SLIDING_WINDOW_SIZE_IN_SECONDS) {
+                // remove from window
+                ThreadMetric prunedData = windowDeque.pollLast();
+                // update blocked time sum for thread in new window
+                if (prunedData != null) {
+                    removeValue(prunedData);
+                }
+            }
+        }
+
+        private void removeValue(ThreadMetric prunedData) {
+            updateValue(prunedData, false);
+        }
+
+        private void addValue(ThreadMetric prunedData) {
+            updateValue(prunedData, true);
+        }
+
+        private void updateValue(ThreadMetric tm, boolean add) {
+            String threadName = tm.name;
+            if (metricSumMap.containsKey(threadName)) {
+                double sign = add ? 1d : -1d;
+                double newVal = metricSumMap.get(threadName) + sign * tm.getValue();
+                if (newVal == 0) {
+                    metricSumMap.remove(threadName);
+                } else {
+                    metricSumMap.put(threadName, newVal);
+                }
+            } else if (add) {
+                metricSumMap.put(threadName, tm.getValue());
+            }
+        }
+
+        public int getCountExceedingThreshold(double threshold) {
+            return (int) metricSumMap.values().stream().filter(val -> val > threshold).count();
+        }
+
+        public double getMaxSum() {
+            return metricSumMap.size() > 0 ? Collections.max(metricSumMap.values()) : 0d;
+        }
+    }
+
+    public static class ThreadMetric {
+        private final String name;
+        private final double value;
+        private final long timeStamp;
+
+        private final String operation;
+
+        public ThreadMetric(String threadName, double val, long timeStamp, String operation) {
+            this.name = threadName;
+            this.value = val;
+            this.timeStamp = timeStamp;
+            this.operation = operation;
+        }
+
+        public String getOperation() {
+            return operation;
+        }
+
+        public long getTimeStamp() {
+            return timeStamp;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public double getValue() {
+            return value;
+        }
+    }
+
+    public static class ThreadAnalysis {
```

Comment:
It might be neater/ more readable moving these to a separate class. There are about 3 additional static classes in this file.

Reply:
Also can the get methods follow the variables and constructor?

Reply:
addressed your comments. plz take a look